lib/internal/v8_prof_polyfill.js

```diff
@@ -26,40 +26,40 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 // Node polyfill
-var fs = require('fs');
-var os = {
+const fs = require('fs');
+const cp = require('child_process');
+const os = {
   system: function(name, args) {
     if (process.platform === 'linux' && name === 'nm') {
       // Filter out vdso and vsyscall entries.
-      var arg = args[args.length - 1];
+      const arg = args[args.length - 1];
       if (arg === '[vdso]' ||
           arg == '[vsyscall]' ||
           /^[0-9a-f]+-[0-9a-f]+$/.test(arg)) {
         return '';
       }
     }
-    return require('child_process').execFileSync(
-        name, args, {encoding: 'utf8'});
+    const cmd = name + ' ' + args.join(' ');
+    return cp.execSync(cmd, {encoding: 'utf8'});
```

Comment:
I don't believe wrapping it in `sh -c` or `bash -c` will work in this instance because the shell script defines a function and then calls it on arguments that will be provided later at the call site. This is the snippet in question: 

```bash
foo() { nm "$@" | (c++filt -p -i || cat) }; foo
``` 

Later it will be invoked as: 

```bash
foo() { nm "$@" | (c++filt -p -i || cat) }; foo -n -f /usr/lib/system/example.dylib
```

Is it possible to perform that wrapping you described in this case?